### DISCUSSION TITLE
How to open a notebook on startup in jupyter-lite's lab?

### DISCUSSION THREAD

**Discussion by MrityunjayBhardwaj** [jupyterlite/jupyterlite#379] [Q&A] [answered]
Is it possible to open a specific notebook on startup in JupyterLab?
I know we can simply use a specific url to open a certain ipynb in retro for e.g. : ```http://localhost:8000/retro/notebooks/?path=pyolite.ipynb```
but i am not sure how to do that in **lab** without manually selecting it from the launcher :thinking: 

Any Suggestions?

**Comment by jtpio** [comment]
Thanks @MrityunjayBhardwaj.

That's right it's not supported in the JupyterLab interface for now.

This might be fixed by https://github.com/jupyterlite/jupyterlite/issues/139. There could also be an endpoint like `/lab/tree?path=pyolite.ipynb` and a plugin to parse the URL to open the file in the main area.

**Comment by bollwyvl** [comment]
+1 to GET params. they are more composable, anyway. they _can_ be given
multiple times, so one could imagine lightweight workspaces, e.g. "open
this document and this picture, and this chat"... Or have some kind of
symbolic language with |, maybe, to allow coarse layout.

**Comment by MrityunjayBhardwaj** [comment]
WoW, I was actually going over the ecosystem of JupyterLab and JupyterLite so that i can create a plugin but before i could even start, @jtpio actually went ahead and did it :open_mouth: :scream:  https://github.com/jupyterlite/jupyterlite/pull/380#issue-1021932246

Damn, I love Open Source Software! Thankyou so much for all the great work you are doing.. :smile: :pray: 

Btw, i also like the idea of taking this even further and allow multiple functionality via GET request and small symbolic language as mentioned by @dsblank

**Comment by jtpio** [comment, marked as answer]
With #380 in, the RTD demo site should now support this: https://jupyterlite.readthedocs.io/en/latest/_static/lab/index.html?path=pyolite.ipynb

And #382 tracks supporting the "Copy Shareable Command" and opening in the Simple Interface with `/doc/tree`.

**Comment by jtpio** [comment]
@MrityunjayBhardwaj some docs about that feature: https://jupyterlite.readthedocs.io/en/latest/user-guide.html#how-can-i-share-a-link-to-a-file

**Reply by steltenpower** [reply]
Too bad the link is dead. Have an alternative?

**Reply by jtpio** [reply]
This should be the new one: https://jupyterlite.readthedocs.io/en/latest/howto/content/share.html

**Reply by steltenpower** [reply]
If that works one could have a browser extension rewrite the HTML showing a list of files, so .ipynb files open up that way into Jupyter Lite. Right?
cc @hfmanson
